xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/utilities/BaseSelect.vue

```diff
@@ -101,30 +102,51 @@ export default {
     sort: {
       type: Boolean,
       default: false,
-    }
+    },
+  },
+
+  computed: {
+    optionsValues () {
+      console.log("optionsValues", this.options.map(option => option.value));
+      console.log("this.options", this.options);
+      console.log("this", this);
+      return this.options.map(option => option.value);
+    },
+
+    selected () {
+      console.log("selected", this.selectedValues.map(value => this.options.find(option => option.value === value)));
+      console.log("this.selectedValues", this.selectedValues);
+      console.log("this", this);
+      return this.selectedValues.map(value => this.options.find(option => option.value === value));
+    },
   },
 
   methods: {
+    isValidOption (option) {
+      return this.optionsValues.includes(option.value);
+    },
+
     isSelected (option) {
-      return this.selected.includes(option);
+      return this.selectedValues.includes(option.value);
     },
 
     select (option) {
-      if (!this.options.includes(option)) { return; }
+      if (!this.isValidOption(option)) { return; }
```

Comment:
See my comment above, the live data configuration could use obsolete values for some query filter and we shouldn't drop these values.